Add delegatesFocus in the Shadow DOM spec.

TakayoshiKochi opened the pull request:
Original proposal:
https://github.com/w3c/webcomponents/blob/gh-pages/proposals/ShadowRoot-delegatesFocus-Proposal.md

Discussion:
https://github.com/w3c/webcomponents/issues/126

Commit:
Add delegatesFocus in the Shadow DOM spec.

Original proposal:
https://github.com/w3c/webcomponents/blob/gh-pages/proposals/ShadowRoot-delegatesFocus-Proposal.md

Discussion:
https://github.com/w3c/webcomponents/issues/126

TakayoshiKochi commented:
@hayatoito PTAL this is continuation from #455.

hayatoito commented on spec/shadow/index.html:
```diff
@@ -1137,6 +1137,20 @@
 
       <section>
+        <h3>Focus</h3>
+      <p>A <a>shadow host</a> can delegate focus to its <a>shadow root</a> by assigning a boolean <a href="#widl-ShadowRootInit-delegatesFocus">delegatesFocus</a> flag to be true in <a href="#idl-def-ShadowRootInit">ShadowRootInit</a> dictionary. If omitted, a <a>shadow host</a> does not delegate focus to its shadow root, and the shadow host itself can be focusable.</p>
```
Could you surround "delegate focus" by `<dfn></dfn>`? And reference it in other places.

TakayoshiKochi replied:
Done.

hayatoito commented on spec/shadow/index.html:
```diff
@@ -1137,6 +1137,20 @@
+        <p>When a shadow host <var>HOST</var> delegates focus, user agent <strong>must</strong> behave as follows.
+          <ol>
+            <li>In <a>sequential focus navigation</a>, <var>HOST</var> itself will be skipped. See the next secition for the formal definition.</li>
+            <li>When <var>HOST</var> is focused by <code>focus()</code> method or <code>autofocus</code> attribute: The first <a>focusable area</a> in the tree order in <var>HOST</var>'s <a>shadow tree</a> gets focus.</li>
```
Tabindex is not used here to determine the first focusable area?

TakayoshiKochi replied:
no, the first one in the focus navigation order of its shadow root.
Replaced the sentence.

Commit:
Fixes review comment from hayatoito.

- Add <dfn></dfn> for "delegates focus" and add references for its
  usage.
- Replaced "The first focusable area in the tree order" with more
  precise wording, referring the next section's focus navigation order.

TakayoshiKochi commented:
@hayatoito PTAL
